language/move-ir/types/src/ast.rs

```diff
@@ -82,6 +82,9 @@ pub struct ModuleDefinition {
     pub name: ModuleName,
     /// the module's dependencies
     pub imports: Vec<ImportDefinition>,
+    /// Explicit declaration of dependencies. If not provided, will be inferred based on given
+    /// dependencies to the IR compiler
+    pub explicit_dependency_declarations: Vec<ModuleDependency>,
```

Comment:
This is an interesting addition to the IR. Some random thoughts: would it be better that
1) We get rid of imports (module aliases).
2) Require dependency declarations.
3) Require expanded names (addr.module_name.func_name for example) everywhere, including in dependency declarations.

This will greatly simplify bytecode generation. If we are worried about the ergonomics of the IR, we could have a higher level representation and add naming and expansion phases to the IR compiler. 

Anyways just something for us to think about...

Reply:
Module aliases are more for readability, but sure we could probably refactor them if we didn't take in CompiledModules

Otherwise, yes. I agree that longterm we would want to switch everything in the IR to using these explicit dependency declarations, and kill off the CompiledModule input